tests/scripts/thread-cert/test_inform_previous_parent_on_reattach.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2021, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import os
+import unittest
+
+import config
+import thread_cert
+
+from pktverify.packet_filter import PacketFilter
+
+# Test description:
+# The purpose of this test is to verify a MED will inform its previous parent when re-attaches to another parent.
+#
+# Initial Topology:
+#
+#  LEADER ----- ROUTER
+#    |
+#   MED
+#
+
+LEADER = 1
+ROUTER = 2
+MED = 3
+
+LONG_CHILD_TIMEOUT = 120
+
+
+class TestReset(thread_cert.TestCase):
+    SUPPORT_NCP = False
+    USE_MESSAGE_FACTORY = False
+
+    TOPOLOGY = {
+        LEADER: {
+            'name': 'LEADER',
+            'mode': 'rdn',
+            'allowlist': [ROUTER, MED]
+        },
+        ROUTER: {
+            'name': 'ROUTER',
+            'mode': 'rdn',
+            'allowlist': [LEADER]
+        },
+        MED: {
+            'name': 'MED',
+            'is_mtd': True,
+            'mode': 'rn',
+            'allowlist': [LEADER],
+            'timeout': LONG_CHILD_TIMEOUT,
+        },
+    }
+
+    def test(self):
+        if 'posix' in os.getenv('OT_CLI_PATH', ''):
+            self.skipTest("skip for posix tests")
```

Comment:
What is the reason for skipping under `posix`?

Reply:
`posix` does not have `INFORM_PREVIOUS_PARENT_ON_REATTACH` enabled by default.

Reply:
Make sense. Thanks.